### Задача 1 
Вероятность того, что стрелок попадет в мишень, выстрелив один раз, равна 0.8. Стрелок выстрелил 100 раз. Найдите вероятность того, что стрелок попадет в цель ровно 85 раз.

In [5]:
from math import factorial
from math import e

Так как количество испытаний маленькое, события независимы, а вероятность события большая используем биномиальное распределение. k = 85, n1 = 100, p = 0.8, q = 1-p = 0.2

In [6]:
def combinations(k, n1, p):
    return (factorial(n1)/(factorial(k) * factorial(n1 - k))) * (p ** k) * ((1 - p) ** (n1 - k))

In [7]:
P = combinations(85, 100, 0.8)

In [8]:
P

0.048061793700746355

Вероятность того, что стрелок попадет в цель ровно 85 раз из ста выстрелов равняется примерно 4.8 %.

### Задача 2 
Вероятность того, что лампочка перегорит в течение первого дня эксплуатации, равна 0.0004. В жилом комплексе после ремонта в один день включили 5000 новых лампочек. Какова вероятность, что ни одна из них не перегорит в первый день? Какова вероятность, что перегорят ровно две?

Количество испытаний большое, вероятность маленькая - будем использовать распределение Пуассона. Где n = 5000, p = 0.0004, lambda = n * p = 2

In [9]:
def combinations(n, p, m):
    return (((n * p)** m)/(factorial(m))) * e**(-n*p)

In [10]:
P = combinations(5000, 0.0004, 0)

In [11]:
P

0.1353352832366127

Вероятность что не перегорит ни одна из лампочек равняется 13.5 %

In [12]:
P = combinations(5000, 0.0004, 2)

In [13]:
P

0.2706705664732254

Вероятность что перегорит ровно две лампочки равняется 27.1 %

### Задача 3 
Монету подбросили 144 раза. Какова вероятность, что орел выпадет ровно 70 раз?

Используем биномиальное распределение. k = 70, n1 = 144, p = 0.5, q = 1-p = 0.5

In [14]:
def combinations(k, n1, p):
    return (factorial(n1)/(factorial(k) * factorial(n1 - k))) * (p ** k) * ((1 - p) ** (n1 - k))

In [15]:
P = combinations(70, 144, 0.5)

In [16]:
P

0.06281178035144776

Вероятность того что орел выпадет 70 раз при 144 бросках примерно равна 6.3 %

### Задача 4 
В первом ящике находится 10 мячей, из которых 7 - белые. Во втором ящике - 11 мячей, из которых 9 белых. Из каждого ящика вытаскивают случайным образом по два мяча. 
Какова вероятность того, что все мячи белые?

P = m/n, где m - число благоприятных исходов, n - общее число исходов. 
Благоприятный исход - когда мы достали 2 белых в первой корзине и два белых во второй(то есть два из 7 в первой и два из 9 во второй). Общий исход когда достали два из 10 в первой и два из 11 во второй. Тогда вероятность того что все мячи белые будет равна произведению вероятности события A (достать два белых мяча из первой корзины) и вероятности события B (достать два белых мяча из второй корзины). То есть P = P(A) * P(B), P(A) = m1/n1, P(B) = m2/n2.

In [19]:
def combinations(k, n1):
    return factorial(n1)/(factorial(k) * factorial(n1 - k))

In [20]:
m1 = combinations(2, 7)
n1 = combinations(2,10)

In [21]:
m1

21.0

In [22]:
n1

45.0

In [24]:
m2 = combinations(2, 9)
n2 = combinations(2,11)

In [25]:
m2

36.0

In [26]:
n2

55.0

In [27]:
m1/n1

0.4666666666666667

In [28]:
m2/n2

0.6545454545454545

In [29]:
(m1/n1) * (m2/n2)

0.3054545454545455

Вероятность того что все мячи белые примерно равна 30.5 %

Какова вероятность того, что ровно два мяча белые?

P = m/n, где m - число благоприятных исходов, n - общее число исходов. Благоприятный исход - когда мы достали 2 белых в первой корзине и ни одного белого во второй(то есть два из 7 в первой и ноль из 9 во второй) либо когда мы достали ни одного белого в первой корзине и два белых во второй(то есть ноль из 7 в первой и два из 9 во второй) либо когда мы достали 1 белый в первой корзине и один белый во второй(то есть один из 7 в первой и один из 9 во второй). Тогда вероятность того что ровно два мяча белые будет равна сумме вероятности события A (достать два белых мяча из первой корзины и ни одного из второй), вероятности события B (достать два белых мяча из второй корзины и ни одного из первой) и вероятности события C (по одному мячу в каждой из корзин - то есть в каждой корзине один белый и один черный). То есть P = P(A) +  P(B) + P(C), P(A) = (m1/n1) * (m11/n11), P(B) = (m2/n2) * (m22/n22), P(C) = (m3 * m333/n3) * (m33 * m3333/n33).

In [30]:
def combinations(k, n):
    return factorial(n)/(factorial(k) * factorial(n - k))

In [40]:
m1 = combinations(2,7)
n1 = combinations(2,10)
m11 = combinations(2, 2)
n11 = combinations(2, 11)

In [41]:
m2 = combinations(2,3)
n2 = combinations(2,10)
m22 = combinations(2, 9)
n22 = combinations(2, 11)

In [43]:
m3 = combinations(1,7)
m333 = combinations(1, 3)
n3 = combinations(2,10)
m33 = combinations(1, 9)
m3333 = combinations(1, 2)
n33 = combinations(2, 11)

In [44]:
P = (m1/n1)*(m11/n11) + (m2/n2)*(m22/n22) + ((m3 * m333)/n3)*((m33 * m3333)/n33)

In [45]:
P

0.20484848484848486

Вероятность того что ровно два мяча белые примерно равна 20.5 %

Какова вероятность того, что хотя бы один мяч белый?

P1 + P = 1 - сумма вероятностей противоположных событий равна 1. Следовательно вероятность что мы достанем хотя бы один белый мяч будет равна: P1 = 1 - P, где P - вероятность того, что мы не достанем ни одного белого мяча.
P = (m1/n1) * (m2/n2)

In [47]:
m1 = combinations(2, 3)
n1 = combinations(2, 10)
m2 = combinations(2, 2)
n2 = combinations(2, 11)

In [48]:
P = (m1/n1) * (m2/n2)

In [49]:
P

0.0012121212121212121

In [50]:
P1 = 1 - P

In [51]:
P1

0.9987878787878788

Вероятность того что мы достанем хотя бы один белый мяч примерно равна 99.9 %